src/libcommongui/logger.cpp

```diff
@@ -47,14 +47,19 @@ static void kdriveLogCatcher(QtMsgType type, const QMessageLogContext &ctx, cons
     if (qtMsgTypeLevel[type] < logger->minLogLevel()) return;
 
     // Create new context
-    const char *fileName = ctx.file;
-    if (fileName) {
-        const char *lastDirSep = strrchr(fileName, '/');
-        if (lastDirSep) {
-            fileName = lastDirSep + 1;
-        }
+    SyncName fileName;
+    if (ctx.file) {
+        const SyncPath filePath(ctx.file);
+        fileName = filePath.filename();
     }
-    QMessageLogContext ctxNew(fileName, ctx.line, ctx.function, ctx.category);
+#ifdef _WIN32
+    // For performance purposes, assume that the file name contains only mono byte chars
+    std::string fileNameStr(fileName.begin(), fileName.end());
```

Comment:
Nice to have: a helper named `toUnsafeStr(const SyncName &name)` define for Windows only.

Reply:
Done